src/main.rs

```diff
@@ -152,31 +152,38 @@ impl HTree {
 // -------------------------------------------------------------------------
 // GIFS
 
-fn create_encoder(file: &mut File, img_size: usize) -> Result<Encoder<&mut File>> {
-    let color_map =
-        &[ 0xFF, 0xFF, 0xFF, // Background RGB
-           0xFF, 0xAA, 0 ];  // Foreground RGB
-
-    let mut encoder: Encoder<&mut File> = Encoder::new(
-        file,
-        img_size as u16,
-        img_size as u16,
-        color_map
-    )?;
-    encoder.set(Repeat::Infinite)?;
-
-    Ok(encoder)
+trait HTreeEncoder {
```

Comment:
There's nothing in this interface that refers to H trees. I think this should just be AnimationEncoder or ImgEncoder, that way when I add more kinds of fractals, we don't have to change the name.

Reply:
Yeah, you're prolly right. Also, this should prolly be composed out of an Encoder rather than implement a trait for it. That way we could also ditch the size passed to add_frame.

Reply:
Sounds good!